docs/manual/determinism-checker.tex

```diff
@@ -152,6 +152,50 @@
   \label{fig-determinism-collections}
 \end{figure}
 
+\subsectionAndLabel{Accessing array elements}{determinism-access-array-elements}
+When an array of type \<@OrderNonDet> (or \<@PolyDet>) is accessed on the rhs of an expression,
+we cannot guarantee that the same element will be returned in every execution. For instance,
+consider an \<@OrderNonDet> array \<a> with elements \code{[5,2,8]}. In one execution,
+\<a[0]> might return \<5> and in another execution, it could return \<8>.
+So we annotate the type of the array access expression (equivalently, the array element) as
```

Comment:
Who's `we`?  This is something the checker does.  You could say that "the type of the array access expression is `@NonDet`".  Please review every "we" in this section and rewrite those that are referring to things the checker does.